### DISCUSSION TITLE
Why latin-1 in response.py and not utf8 ?

### DISCUSSION THREAD

**Discussion by svsdval** [encode/starlette#2755] [Q&A] [answered]
I tried to set cookies in utf8
```python
response = RedirectResponse(url=ref_url, status_code=status.HTTP_303_SEE_OTHER, headers=request.headers)
response.set_cookie(key="success", value='Данные обновлены',max_age=5)
response.set_cookie(key="errors", value=['Ошибка 1','Ошибка 2', 'Ошибка 3'],max_age=5)
```
but got an error:
`
UnicodeEncodeError: 'latin-1' codec can't encode characters in position 9-14: ordinal not in range(256) fastapi utf8
`
Then i replaced all "latin-1" to "utf8" in starlette/responses.py, and checked and it works fine =) 
In browser 
![image](https://github.com/user-attachments/assets/525bd8bb-a196-4624-bea7-b8555b74ae37)

So... Why limit responses to "latin-1" encoding? and not use "utf8" ?, Or do it in the right way? Or is there some way to change the encoding?

**Comment by svsdval** [comment]
I don't know how correct this solution is, I dumped the cookies variables into json and encoded it into base64, and then decoded it and loaded with json

**Comment by Kludex** [comment, marked as answer]
We are following the HTTP RFCs.

See previous attempt to change it here: https://github.com/encode/starlette/pull/1236
